src/Microsoft.AspNet.WebUtilities/QueryHelpers.cs

```diff
@@ -20,8 +20,7 @@ public static class QueryHelpers
         /// <returns>The combined result.</returns>
         public static string AddQueryString([NotNull] string uri, [NotNull] string name, [NotNull] string value)
         {
-            bool hasQuery = uri.IndexOf('?') != -1;
-            return uri + (hasQuery ? "&" : "?") + UrlEncoder.Default.UrlEncode(name) + "=" + UrlEncoder.Default.UrlEncode(value);
+            return AddQueryString(uri, new Dictionary<string, string> { { name, value } });
```

Comment:
Perf: dictionaries aren't cheep. Do not create a new one just to add one key-value pair.